pkg/types/baremetal/validation/platform_test.go

```diff
@@ -167,6 +167,26 @@ func TestValidatePlatform(t *testing.T) {
 				BootstrapOSImage("192.168.111.1/images/qemu.x86_64.qcow2.gz?sha256=3b5a882c2af3e19d515b961855d144f293cab30190c2bdedd661af31a1fc4e2f").build(),
 			expected: "baremetal.BootstrapOSImage: Invalid value:.*: the URI provided:.*is invalid",
 		},
+		{
+			name: "invalid_clusterosimage_not_qcow2",
+			platform: &baremetal.Platform{
```

Comment:
Hi Ian! What about using the fluent interface, I think it helps in improving the readability?

Reply:
Well it took me way longer to write the test but it's there now :).  It's too "magical" for me, but that's just my opinion.